add islocked filter to SeachIssuesRequest (#2623)
octokit/octokit.net: docs/search.md

request.State = ItemState.All;
// or to just search closed issues
request.State = ItemState.Closed;

> // you can filter by the is qualifier
> // the enum IssueIsQualifier contains the supported values
> // here is how you will can filter with issue and locked

Reviewer: ```suggestion
// you can filter by the "Is" qualifier
// the enum IssueIsQualifier contains the supported values
// you can filter for locked issues like this:
```
